backend/rpc-impl/src/main/java/org/sonarsource/sonarlint/core/rpc/impl/SonarLintRpcServerImpl.java

```diff
@@ -318,6 +335,10 @@ public StorageService getIssueStorageService() {
     return getInitializedApplicationContext().getBean(StorageService.class);
   }
 
+  public SonarLintDatabase getDatabase() {
+    return getInitializedApplicationContext().getBean(SonarLintDatabase.class);
```

Comment:
As discussed, this is exposing the internals. The approach is the same as before, but if we have time I would like to do it differently